server/services/validator_service.py

```diff
@@ -129,6 +129,48 @@ def unlock_tasks_after_validation(validated_dto: UnlockAfterValidationDTO) -> Ta
         return task_dtos
 
     @staticmethod
+    def stop_validating_tasks(stop_validating_dto: StopValidationDTO) -> TaskDTOs:
```

Comment:
I think we should really refactor this, as it's a 90% cut and paste job.  So it means bugs will inevitably creep in as Unlock changes but Stop doesn't etc.  So common code can be called by both Unlock and Stop